strawberry/extensions/context.py

```diff
@@ -73,10 +80,16 @@ def get_hook(self, extension: SchemaExtension) -> Optional[WrappedHook]:
 
         if hook_fn:
             if inspect.isgeneratorfunction(hook_fn):
-                return WrappedHook(extension, hook_fn(extension), False)
+                hook_fn = types.MethodType(
+                    contextlib.contextmanager(hook_fn), extension
+                )
+                return WrappedHook(extension, hook_fn, False)
```

Comment:
Maybe it would be better to pass `True/False` as kwarg, it's a bit more readable

Reply:
yup, or all of them as kwargs!